### PR TITLE
wxDataView optimizations.

### PR DESCRIPTION
**Pull request by erikhugintech**
Redoing this pullrequest from a branch I don't need to update.

As versatile as wxDataViewCtrl is, it falls short on large data sets, mainly due to inefficient use of sorting. This commit starts to address this.

Resort everything on thaw is removed. Recursive resorts are now applied only to open nodes. Node list sort state is retained, and node lists are only sorted when strictly necessary. Inserts into sorted lists are done at the correct position instead of appending and resorting.

There should be ample comments in the diff, explaining rationales.

With this patch we're adding a few million items to a sorted wxDataViewCtrl without problematic performance. Without this patch, the same application pretty much froze for some hours while building the same item tree.

I've tried hard to follow your coding style guidelines, however as this style differs significantly from my personal style, it's quite possible that I've slipped here and there.

### CODE REVIEW

File: src/generic/datavgen.cpp [hunk 1]
```diff
@@ -1636,39 +1618,207 @@ void wxDataViewTreeNode::InsertChild(wxDataViewTreeNode *node, unsigned index)
     if (!m_branchData)
         m_branchData = new BranchNodeData;
 
-    m_branchData->children.Insert(node, index);
+    g_column = m_window->GetSortColumn();
+    g_model = m_window->GetModel();
+    g_asending = m_window->IsAscendingSort();
```
Comment (adrientetar) [inline src/generic/datavgen.cpp]: typo: g_ascending

**Comment by erikhugintech** [comment]
Commited fix

**Comment by Prince-David** [comment]
I saw this PR last day, and I hope there is someone who have time to test it ...

It seems interesting 👍

**Comment by vadz** [comment]
@erikhugintech Sorry for getting back to this so late, but I'm going to look at/test this and merge it soon, finally. Just one question before I look into the details: why was the code suspending sorting while the control is frozen removed? Was this done intentionally, i.e. is it really not needed any longer (and, if so, why?)?

**Comment by erikhugintech** [comment]
Been a while since I looked at it, but as I recall it the unfreeze sort isn't needed as branches are sorted on demand when opening, and kept sorted by inserting at correct points. An additional sort when unfreezing would just sort branches that are either already sorted or not visible to the end user.

**Review by vadz — COMMENTED** [review]
I've addressed the few things that didn't seem ideal to me, as well as a bunch of other minor ones (e.g. style/indentation fixes) in the new version of the changes included in the PR #642. Please have a look at it if you can and let me know if you have any remarks. TIA!

**Comment by vadz** [comment]
Closed as superseded by #642, which should be merged soon.

Thanks again for your work on this!

**Pull request closed without merging**